rcon/commands.py

```diff
@@ -156,19 +157,26 @@ def with_connection(self) -> HLLConnection:
                     raise TimeoutError()
                 c += 1
                 time.sleep(1)
-                with self.mu:
-                    idle_len = len(self.idles)
 
-            with self.mu:
-                conn = self.idles.pop()
-                logger.debug("connection appeared in pool: %s, acquiring it", conn.id)
+                if not self.mu.acquire(timeout=30):
```

Comment:
why not putting this timeout in the context manager directly?

Reply:
I don't know a way of passing a timeout to acquire when using the context manager 🤔 I mean, without writing my own contextmanager for that :D